tests/Backend/SOX/SOXTokensTest.php

```diff
@@ -72,15 +86,39 @@ public function testTokensVisibility(Client $client): void
         }
     }
 
-    public function testCannotRefreshCanManageProtectedBranchTokenEvenSelf(): void
+    /**
+     * @dataProvider developerAndReviewerClientProvider
+     * @dataProvider prodManagerClientProvider
+     * @dataProvider priviledgedTokenClientProvider
+     */
+    public function testCannotRefreshCanManageProtectedBranchTokenEvenSelf(Client $client): void
     {
-        $client = $this->getDefaultBranchStorageApiClient();
         $tokens = new Tokens($client);
         $this->expectExceptionCode(400);
         $this->expectExceptionMessage('Token with canManageProtectedDefaultBranch privilege cannot be refreshed');
+        // getDefaultBranchTokenId = priviledged token
         $tokens->refreshToken($this->getDefaultBranchTokenId());
     }
 
+    /**
+     * @dataProvider developerAndReviewerClientProvider
+     * @dataProvider prodManagerClientProvider
+     */
+    public function testRefreshToken(Client $client): void
+    {
+        $tokens = new Tokens($client);
+        $token = $tokens->createToken((new TokenCreateOptions()));
+
+        sleep(1);
+
+        $token = $tokens->refreshToken($token['id']);
+
+        $this->assertGreaterThan(
+            (new DateTime($token['created']))->getTimestamp(),
+            (new DateTime($token['refreshed']))->getTimestamp()
+        );
```

Comment:
tady testuješ, že klientem možeš refreshnůt token cos vytvořil, měl bys otestovat, že ten token može refreshnůt sám sebe.
A taky bych to zakázal https://keboolaglobal.slack.com/archives/C05BK5V8N1Z/p1689153668906609